### PR TITLE
Hook up cutscenes

### PR DESCRIPTION
**Pull request by SheridanR**

### CODE REVIEW

File: src/actladder.cpp [hunk 1]
```diff
@@ -488,19 +499,61 @@ void actWinningPortal(Entity* my)
 						}
 						strcpy((char*)net_packet->data, "WING");
 						net_packet->data[4] = victory;
+						net_packet->data[5] = cutscene;
 						net_packet->address.host = net_clients[c - 1].host;
 						net_packet->address.port = net_clients[c - 1].port;
-						net_packet->len = 8;
+						net_packet->len = 5;
```
Comment (WALLOFJUSTICE) [inline src/actladder.cpp:505]: should be `net_packet->len = 6;` due to cutscene var

File: src/actladder.cpp [hunk 2]
```diff
@@ -488,19 +499,61 @@ void actWinningPortal(Entity* my)
...
-				introstage = 5; // prepares win game sequence
-				fadeout = true;
-				if ( !intro )
-				{
-					pauseGame(2, false);
-				}
+
+	            if (cutscene == 1) { // classic herx ending
+	                switch (stats[clientnum]->playerRace) {
+	                default:
```
Comment (WALLOFJUSTICE) [inline src/actladder.cpp:512]: TIL the concept of putting `default:` anywhere but the bottom 😄

File: src/actladder.cpp [hunk 3]
```diff
@@ -786,56 +857,81 @@ void Entity::actMidGamePortal()
...
+	                case RACE_SKELETON:
+	                case RACE_VAMPIRE:
+	                case RACE_SUCCUBUS:
+	                case RACE_INCUBUS:
+	                    MainMenu::beginFade(MainMenu::FadeDestination::BaphometMidpointEvil);
+	                    break;
+	                }
+	            }
+
+                movie = true;
```
Comment (WALLOFJUSTICE) [inline src/actladder.cpp:933]: Reminder to check Entity::isMobile to make sure monsters don't move around in cutscenes

File: src/entity.cpp [hunk 4]
```diff
@@ -5766,13 +5766,9 @@ bool Entity::isMobile()
...
 
-	if ( behavior == &actMonster && 
-		(introstage == 9
-		|| introstage == 11 + MOVIE_MIDGAME_BAPHOMET_HUMAN_AUTOMATON
-		|| introstage == 11 + MOVIE_MIDGAME_BAPHOMET_MONSTERS
-		|| introstage == 11 + MOVIE_MIDGAME_HERX_MONSTERS) )
+	if ( behavior == &actMonster && MainMenu::isCutsceneActive() )
 	{
-		return false; // mid-game crawls.
+		return false;
```
Comment (WALLOFJUSTICE) [inline src/entity.cpp:5771]: Verify this is working correctly? I.e don't get slaughtered by demons in hell while watching cutscene in network lobby

File: src/net.cpp [hunk 5]
```diff
@@ -351,37 +351,20 @@ bool messagePlayerColor(int player, Uint32 type, Uint32 color, char const * cons
...
-		    }
-		    if ( !disable_messages )
-		    {
-			    players[player]->messageZone.addMessage(color, buf);
-		    }
-		    ptr += c + 1;
-		} while (!end);
+	    printlog("%s\n", str);
+	    newString(&messages, color, str);
+	    while ( list_Size(&messages) > MESSAGE_LIST_SIZE_CAP )
```
Comment (WALLOFJUSTICE) [inline src/net.cpp:363]: Do messages process / expire for disconnected players? Like reverting using `/splitscreen` to singleplayer. Used to get backed up

File: src/entity.cpp (hunk 4 above)
Comment (SheridanR) [reply src/entity.cpp:5771]: It works fine but I added another if check to actMonster(). it returns immediately `if (movie || MainMenu::isCutsceneActive())`

File: src/net.cpp (hunk 5 above)
Comment (SheridanR) [reply src/net.cpp:363]: I don't think I fixed that issue. OTOH that's not possible in a normal game.

File: src/actladder.cpp (hunk 1 above)
Comment (SheridanR) [reply src/actladder.cpp:505]: Thanks, I was afraid I messed up one of those.

**Pull request merged**